molgenis-ontology/src/main/java/org/molgenis/ontology/controller/SortaServiceController.java

```diff
@@ -318,21 +342,21 @@ public OntologyServiceResult matchResult(@RequestBody Map<String, Object> reques
 			String inputTermIdentifier = request.get(MatchingTaskContentEntityMetaData.IDENTIFIER).toString();
 			Entity matchingTaskEntity = dataService.findOne(MatchingTaskEntityMetaData.ENTITY_NAME,
 					new QueryImpl().eq(MatchingTaskEntityMetaData.IDENTIFIER, entityName));
-			Entity entity = dataService.findOne(entityName,
+			Entity inputEntity = dataService.findOne(entityName,
 					new QueryImpl().eq(MatchingTaskContentEntityMetaData.IDENTIFIER, inputTermIdentifier));
 
-			if (matchingTaskEntity == null || entity == null)
-				return new OntologyServiceResult("entityName or inputTermIdentifier is invalid!");
+			if (matchingTaskEntity == null || inputEntity == null)
+				return new SortaServiceResponse("entityName or inputTermIdentifier is invalid!");
 
-			return ontologyService.searchEntity(matchingTaskEntity.getString(MatchingTaskEntityMetaData.CODE_SYSTEM),
-					entity);
+			return new SortaServiceResponse(inputEntity, sortaService.findOntologyTermEntities(
+					matchingTaskEntity.getString(MatchingTaskEntityMetaData.CODE_SYSTEM), inputEntity));
 		}
-		return new OntologyServiceResult("Please check entityName, inputTermIdentifier exist in input!");
+		return new SortaServiceResponse("Please check entityName, inputTermIdentifier exist in input!");
```

Comment:
Can you make it more user friendly? "<entityName> or <inputTermIdentifier> is invalid!"